internal/jwk/key.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright © 2025 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+// Package jwk provides JSON Web Key (JWK) structure and methods to decode it to public and private keys.
+package jwk
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/big"
+)
+
+const typeRSA = "RSA"
+
+var supportedKeyTypes = map[string]struct{}{
+	typeRSA: {},
+}
+
+// Key defines JSON Web Key structure.
+type Key struct {
+	Alg string `json:"alg"`           // algorithm
+	Crv string `json:"crv,omitempty"` // curve - for EC keys
+	D   string `json:"d"`             // private exponent
+	DP  string `json:"dp"`            // d mod (p-1)
+	DQ  string `json:"dq"`            // d mod (p-1)
+	E   string `json:"e"`             // public exponent
+	K   string `json:"k,omitempty"`   // symmetric key
+	Kid string `json:"kid"`           // Key ID
+	Kty string `json:"kty"`           // Key Type
+	N   string `json:"n"`             // modulus
+	P   string `json:"p"`             // prime factor 1
+	Q   string `json:"q"`             // prime factor 2
+	QI  string `json:"qi"`            // q^(-1) mod p
+	Use string `json:"use"`
+	X   string `json:"x,omitempty"` // x coordinate - for EC keys
+	Y   string `json:"y,omitempty"` // y coordinate - for EC keys
+}
+
+// DecodePublicKey decodes Key to public key.
+func (j *Key) DecodePublicKey() (crypto.PublicKey, error) {
+	if _, ok := supportedKeyTypes[j.Kty]; !ok {
+		return nil, fmt.Errorf("unsupported key type %s", j.Kty)
+	}
+
+	var result interface{}
+
+	if j.Kty == typeRSA {
+		if j.N == "" || j.E == "" {
+			return nil, errors.New("malformed JWK RSA key")
```

Comment:
Would you consider providing the reason why the JWK RSA key is malformed? I think it would help in better understanding the root cause of the issue.
p.s.: I think it's fine to improve error messages in a separate PR if needed.